External/FEXCore/Source/Interface/Core/CPUID.cpp

```diff
@@ -25,89 +28,321 @@ CPUIDEmu::FunctionResults CPUIDEmu::Function_0h() {
+    (0 <<  2) | // Usermode instruction prevention
+    (0 <<  3) | // Protection keys for user mode pages
+    (1 <<  4) | // OS protection keys
+    (0 <<  5) | // waitpkg
+    (0 <<  6) | // AVX512_VBMI2
+    (0 <<  7) | // CET shadow stack
+    (0 <<  8) | // GFNI
+    (0 <<  9) | // VAES
+    (0 << 10) | // VPCLMULQDQ
+    (0 << 11) | // AVX512_VNNI
+    (0 << 12) | // AVX512_BITALG
+    (0 << 13) | // Intel Total Memory Encryption
+    (0 << 14) | // AVX512_VPOPCNTDQ
+    (0 << 15) | // Reserved
+    (0 << 16) | // 5 Level page tables
+    (0 << 17) | // MPX MAWAU
+    (0 << 18) | // MPX MAWAU
+    (0 << 19) | // MPX MAWAU
+    (0 << 20) | // MPX MAWAU
+    (0 << 21) | // MPX MAWAU
+    (0 << 22) | // RDPID Read Processor ID
+    (0 << 23) | // Reserved
+    (0 << 24) | // Reserved
+    (0 << 25) | // CLDEMOTE
+    (0 << 26) | // Reserved
+    (0 << 27) | // MOVDIRI
+    (0 << 28) | // MOVDIR64B
+    (0 << 29) | // Reserved
+    (0 << 30) | // SGX Launch configuration
+    (0 << 31);  // Reserved
+
+  Res.edx =
+    (0 <<  0) | // Reserved
+    (0 <<  1) | // Reserved
+    (0 <<  2) | // AVX512_4VNNIW
+    (0 <<  3) | // AVX512_4FMAPS
+    (0 <<  4) | // Fast Short Rep Mov
+    (0 <<  5) | // Reserved
+    (0 <<  6) | // Reserved
+    (0 <<  7) | // Reserved
+    (0 <<  8) | // AVX512_VP2INTERSECT
+    (0 <<  9) | // Reserved
+    (0 << 10) | // VERW clears CPU buffers
+    (0 << 11) | // Reserved
+    (0 << 12) | // Reserved
+    (0 << 13) | // Reserved
+    (0 << 14) | // SERIALIZE instruction
+    (0 << 15) | // Reserved
+    (0 << 16) | // Reserved
+    (0 << 17) | // Reserved
+    (0 << 18) | // Intel PCONFIG
+    (0 << 19) | // Intel Architectural LBR
+    (0 << 20) | // Intel CET
+    (0 << 21) | // Reserved
+    (0 << 22) | // Reserved
+    (0 << 23) | // Reserved
+    (0 << 24) | // Reserved
+    (0 << 25) | // Reserved
+    (0 << 26) | // Reserved
+    (0 << 27) | // Reserved
+    (0 << 28) | // L1D Flush
+    (0 << 29) | // Arch capabilities
+    (0 << 30) | // Reserved
+    (0 << 31);  // Reserved
+
   return Res;
 }
 
 // Highest extended function implemented
 CPUIDEmu::FunctionResults CPUIDEmu::Function_8000_0000h() {
   CPUIDEmu::FunctionResults Res{};
-  Res.Res[0] = 0x8000001F;
+  Res.eax = 0x8000001F;
   return Res;
 }
 
 // Extended processor and feature bits
 CPUIDEmu::FunctionResults CPUIDEmu::Function_8000_0001h() {
   CPUIDEmu::FunctionResults Res{};
-  Res.Res[2] = ~0U; // Let's say we support every feature for fun
-  Res.Res[3] = ~0U; // Let's say we support every feature for fun
-
-  Res.Res[3] &= ~(
-    (1 << 6)  | // Remove SSE4a
-    (1 << 11) | // Remove XOP
-    (1 << 16)   // Remove FMA4
-    );
+
+  Res.eax = 0 | // Stepping
+    (0 << 4) | // Model
+    (0 << 8) | // Family ID
+    (0 << 12) | // Processor type
+    (0 << 16) | // Extended model ID
+    (0 << 20); // Extended family ID
+
+  Res.ecx =
+    (1 <<  0) | // LAHF/SAHF
+    (1 <<  1) | // 0 = Single core product, 1 = multi core product
```

Comment:
More of a question than a code review: 

Should we be setting this to 0 when we set the number of threads to zero?

Reply:
Without hardware testing this on a real single core product I don't really want to fuss with it.
Apparently the last one released from Intel is a Sandy Bridge Celeron G470 CPU back in 2013.